src/convertToObject.js

```diff
@@ -6,7 +6,36 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitCleanProperties = sourceString
+    .split(';')
+    .map((dirtyProperty) => dirtyProperty.trim());
+
+  const validPropertyStrings = splitCleanProperties.filter((property) => {
+    return property !== '';
+  });
+
```

Comment:
```suggestion
  const validPropertyStrings = splitCleanProperties.filter((property) => property);

```

And you can just add this `filter` to previous chain

Reply:
Edited filter predicate, though linter still does not like as a one-liner.

About chaining filter. I talked to Danylo Hryhorets about this, and he advised against it in [this thread](https://mate.academy/chats/222864?selectedMessageId=2994558&threadId=2993849). And I'm kinda tired of redoing it for the third time, to be honest...